NitroxClient/MonoBehaviours/Gui/Settings/SettingsOptions.cs

```diff
@@ -0,0 +1,103 @@
+﻿using NitroxClient.Unity.Helper;
+using NitroxModel.Logger;
+using Steamworks;
+using UnityEngine;
+using UnityEngine.Events;
+using UnityEngine.UI;
+
+namespace NitroxClient.MonoBehaviours.Gui.Settings {
+    public class SettingsOptions {
+        public void AddInputFieldOption(int tabIndex, string label, string placeholder, string value, uGUI_OptionsPanel uGUI_Options, UnityAction<string> callback = null) {
+            uGUI_MainMenu uGUI_Main = uGUI_MainMenu.main;
+            uGUI_TabbedControlsPanel uGUI_Control = uGUI_Main.GetComponentInChildren<uGUI_TabbedControlsPanel>();
+            GameObject InputOption = new GameObject();
+            GameObject input_field = Object.Instantiate(uGUI_Options.keyRedemptionPrefab.transform.GetChild(1).gameObject);
+            GameObject caption = Object.Instantiate(uGUI_Options.toggleOptionPrefab.transform.GetChild(0).GetChild(2).gameObject);
```

Comment:
Will these indices stay the same if UWE changes the layout? Can we change this to use paths (via `.Find`)?